### PR TITLE
refactor(core): add host directive definitions validation

### PR DESCRIPTION
**Pull request by crisbeto**
Adds some logic to ensure that host directives are configured correctly.

### CODE REVIEW

File: packages/core/src/render3/features/host_directives_feature.ts [hunk 1]
```diff
@@ -60,9 +61,17 @@ function findHostDirectiveDefs(
     hostDirectiveDefs: HostDirectiveDefs): void {
   if (currentDef.hostDirectives !== null) {
     for (const hostDirectiveConfig of currentDef.hostDirectives) {
-      const hostDirectiveDef = getDirectiveDef(hostDirectiveConfig.directive)!;
+      const {
+        directive: hostDirectiveReference,
+        inputs: exposedInputs,
+        outputs: exposedOutputs,
+      } = hostDirectiveConfig;
```
Comment (pkozlowski-opensource) [inline packages/core/src/render3/features/host_directives_feature.ts:68]: nit: move this destructure inside `ngDevMode` ? Or don't destruct and just access appropriate fields as needed?

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM overall. Left a couple of nit comments that you can safely ignore.

**Comment by crisbeto** [comment]
Feedback has been addressed.

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: public-api 
Reviewed-for: fw-core

**Review by atscott — APPROVED** [review]
reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 50f8928d56bdbc1e65dfd5127ec00d8d934d2432.

**Pull request closed without merging**